test/colab/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM --platform=linux/amd64 python:3.7.13-slim-buster
+
+RUN apt-get update \
+    && apt-get install --no-install-recommends --assume-yes \
+    libgdal-dev r-base build-essential libglib2.0-0 libglib2.0-dev \
+    libsndfile1 libsndfile1-dev cmake libgcc1 gcc-7-base gcc-8-base \
+    g++ g++-7 gcc python-apt libgirepository1.0-dev \
+    libcairo2-dev libjpeg-dev libgif-dev python3-cairo-dev python3-dev \
+    libbz2-dev libreadline-dev libssl-dev zlib1g-dev libsqlite3-dev wget \
+    curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev \
+    python-apt-common python3-apt python3-distutils-extra python-apt \
+    vim git \
+    && curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python3 - -y \
+    && apt-get autoremove -y
+
+ENV CPLUS_INCLUDE_PATH=/usr/include/gdal
+ENV C_INCLUDE_PATH=/usr/include/gdal
+
+COPY build/colab-test/requirements-fixed.txt /tmp/requirements-fixed.txt
+RUN --mount=type=cache,mode=0755,target=/root/.cache/pip pip install --no-dependencies -r /tmp/requirements-fixed.txt
+WORKDIR /usr/src/app
+
+CMD ["./test/colab/test.sh"]
```

Comment:
This file is not actually in the image right? Can we not set CMD and instead pass this to the container when we run it.

Reply:
Not sure I get what the difference would be?

Reply:
The difference is that we don't bake anything in the image that is not related to colab.

For instance now if I decide to rename this file, I need to rebuild the whole image.

Reply:
CMD provides a default which can be overwritten.  
If you want to rename the file you can update the docker run command and add `./test/colab/new-test-file.sh` to the end of it to point to the new file, as you say.  

The point of a default is to make sense in most cases, and for now we're only planning to use this image for this one purpose.

Reply:
I think it is better to keep the image agnostic from the test code instead of being tightly coupled. I personally think of this image as a CI executor, just like the ones we use in CircleCI - it has the right environment for us to add the test code and execute. I see them as two separated components we will be mainteining and integrating separately.

We will have to break this tight coupling when we start building the image nightly and likely move to a separate repository.

It is also quite confusing that the CMD references a file that isn't actually in the image.

I am ok with keeping it as it is in the context of this PR, but I feel like we will have to break that coupling soon after.

Reply:
I don't believe how we build the image will affect how we use it. When we automate the creation of the image, it will still only be used for this test, initially.  
If down the road we have another usecase for the image and end up using it as a general purpose executor, then sure, we might have to decouple the image from the tests, and maybe even create a different image building on top of this one, like we do for the executors, but let's not optimise prematurely.

For now this is a single purpose image, let's just focus on making sure it does what it's supposed to do well first. 😃 

Let's talk about the automated build process separately, as I'm sure we'll disagree on that too!